replica/table.cc

```diff
@@ -437,6 +438,30 @@ void table::enable_off_strategy_trigger() {
 
 future<>
 table::do_add_sstable_and_update_cache(sstables::shared_sstable sst, sstables::offstrategy offstrategy) {
+    if (sst->requires_view_building()) {
+        auto [it, inserted] = _sstables_staging.emplace(sst->generation(), sst);
+        if (!inserted) {
+            on_internal_error(tlogger, format("could not add staging sstable: generation {} already exists", sst->generation()));
+        }
+
+        // It is possible that after restart, the previously cloned sstable
+        // still exists in the main table directory, so we're done.
+        // Nothing more to do here.
+        if (_sstables->contains(sst)) {
+            co_return;
+        }
```

Comment:
Ah, so cloning keeps the same generation? Perhaps cloning should create a new generation. Then we can simply ignore duplicates.

Reply:
This can be done.  The only downside I can think of is that a restart before view building finished will cause a wave cloning since we currently don't have a way to seal the sstable in the staging directory AND clone it to the table base directory atomically. We can have one similar to the way we implemented `delete_atomically` using a an intent file but it seems like an overkill to me, and re-adding the sstables after restart should be no worse than today's situation.

Reply:
Done in v6